File: FelipezConde/QuestionnaireLanguage/AST/Types/StringType.cs

namespace AST.Types
{
    public class StringType : Type
    {
        public override bool IsString()
        {
            return true;
        }

        public override bool IsEqual(Type type)
        {
            return type.IsString();
        }

        public override T Accept<T>(Visitors.ITypeVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public override string ToString()
        {
            return "string";
        }
    }
}

File: Fugazi/src/org/fugazi/ql/ast/type/StringType.java
package org.fugazi.ql.ast.type;

public class StringType extends Type {

    public StringType() {
        super();
    }

    @Override
    public String toString() {
        return "String";
    }

    public <T> T accept(ITypeVisitor<T> visitor) {
        return visitor.visitStringType(this);
    }
}